src/map/lua/lua_baseentity.cpp

```diff
@@ -12423,6 +12423,27 @@ inline int32 CLuaBaseEntity::removeAllManeuvers(lua_State* L)
 
     return 0;
 }
+/************************************************************************
+*  Function: setMobLevel()
+*  Purpose : Updates the monsters level and recalculates stats
+*  Example : mob:setMobLevel(125)
+*  Notes   : CalculateStats will refill mobs hp/mp as well
+************************************************************************/
+
+inline int32 CLuaBaseEntity::setMobLevel(lua_State *L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB);
+
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+
+    CMobEntity * PMob = (CMobEntity*)m_PBaseEntity;
```

Comment:
Should still be checking for nullptr here to prevent crashing in release? Either way, we should stay away from c-style casting on complex types.

can either use dynamic_cast<CMobEntity*> to check for nullptr or static_cast<CMobEntity*> if we don't care.

Reply:
ah rightm bad copy pasta. will fix where I got it from too